test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -467,6 +467,52 @@ long GetAggregatedValue(Metric metric)
             }
         }
 
+        [Fact]
+        public void DuplicateInstrumentRegistration_WithViews_OneInstrument_MultipleViews_OneStream()
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}");
+            var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddView((instrument) =>
+                {
+                    return new MetricStreamConfiguration { TagKeys = new[] { "key1" } };
+                })
+                .AddView((instrument) =>
+                {
+                    return new MetricStreamConfiguration { TagKeys = new[] { "key2" } };
+                })
+                .AddInMemoryExporter(exportedItems);
+
+            using var meterProvider = meterProviderBuilder.Build();
+
+            var instrument1 = meter.CreateCounter<long>("name");
```

Comment:
`var instrument2 = meter.CreateCounter<long>("name");`
If we have another instrument like the above, then it should be contributing to the same 2 metric streams as the instrument1, but in the current fix, it'll only contribute to one.